### PR TITLE
admin: Add account endpoint.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Part of #329

### CODE REVIEW

File: server/db/driver/pg/accounts.go [hunk 1]
```diff
@@ -53,12 +54,18 @@ func (a *Archiver) Accounts() ([]*db.Account, error) {
...
 		a := new(db.Account)
-		err = rows.Scan(&a.AccountID, &a.Pubkey, &a.FeeAddress, &a.FeeCoin, &a.BrokenRule)
+		err = rows.Scan(&accountID, &pubkey, &a.FeeAddress, &feeCoin, &brokenRule)
 		if err != nil {
 			return nil, err
 		}
+		copy(a.AccountID[:], accountID)
+		a.Pubkey = dex.Bytes(pubkey)
+		a.FeeCoin = dex.Bytes(feeCoin)
+		a.BrokenRule = account.Rule(brokenRule)
```
Comment (JoeGruffins) [inline server/db/driver/pg/accounts.go:68]: Whoops:
`failed to retrieve accounts: sql: Scan error on column index 3, name "fee_coin": unsupported Scan, storing driver.Value type <nil> into type *dex.Bytes`
I suppose it is because sql doesn't work well with custom types? Have removed all custom types from the `Scan` just in case.

Comment (chappjc) [reply server/db/driver/pg/accounts.go:68]: Custom types need either (1) built-in underlying type, or (2) define Scan and Value methods to satisfy sql.Scanner and sql.Valuer interfaces.  We have those define for many type.  I suggest looking at those for examples. e.g. for `AccountID`: https://github.com/decred/dcrdex/blob/95d8c3ffd08d6c9c87d7033b25b73a6c14f03027/server/account/account.go#L41-L59, which does the copy that you are doing manually now.

However, the error here indicates that `dex.Bytes`, which is just `[]byte` doesn't handle NULL values in the table.  Note that the NULL issue is so common that there are standard types like `sql.NullString` and `sql.NullInt64` for this reason.  For byte slices, I think it should just work with NULLs.  How do you get this error (what code and test)?

Comment (chappjc) [reply server/db/driver/pg/accounts.go:68]: I can repro the error by changing `getAccount` to use `dex.Bytes` instead of `[]byte`.  I'm OK with Scanning into a `[]byte` and just converting like you have.  But we can definitely scan into an account.AccountID directly because we have implemented sql.Scanner for that.

File: server/db/driver/pg/accounts_online_test.go [hunk 2]
```diff
@@ -89,6 +90,14 @@ func TestAccounts(t *testing.T) {
 		t.Fatal("accounts has unexpected data")
 	}
 
+	anAcct, err := archie.AccountInfo(accts[0].AccountID)
```
Comment (chappjc) [inline server/db/driver/pg/accounts_online_test.go:93]: Let's also test `PayAccount` before `PayAccount` to test the null cases.

File: server/db/driver/pg/accounts.go [hunk 3]
```diff
@@ -67,6 +74,22 @@ func (a *Archiver) Accounts() ([]*db.Account, error) {
...
+	stmt := fmt.Sprintf(internal.SelectAccountInfo, a.tables.accounts)
+	acct := new(db.Account)
+	var accountID, pubkey, feeCoin []byte
+	var brokenRule byte
+	if err := a.db.QueryRow(stmt, aid).Scan(&accountID, &pubkey, &acct.FeeAddress, &feeCoin, &brokenRule); err != nil {
+		return nil, err
+	}
+	copy(acct.AccountID[:], accountID)
+	acct.Pubkey = dex.Bytes(pubkey)
+	acct.FeeCoin = dex.Bytes(feeCoin)
```
Comment (chappjc) [inline server/db/driver/pg/accounts.go:88]: The explicit `dex.Bytes` conversion is superfluous given the definition `type dex.Bytes []byte`.  That is, `a.Pubkey = pubkey` is good.

File: server/admin/api.go [hunk 4]
```diff
@@ -156,3 +158,21 @@ func (s *Server) apiAccounts(w http.ResponseWriter, _ *http.Request) {
 	}
 	writeJSON(w, accts)
 }
+
+// apiAccountInfo is the handler for the '/account/{account id}' API request.
+func (s *Server) apiAccountInfo(w http.ResponseWriter, r *http.Request) {
+	acct := strings.ToLower(chi.URLParam(r, accountNameKey))
```
Comment (chappjc) [inline server/admin/api.go:164]: Seems like the ToLower call is not needed.  hex.DecodeString seems to be find with mixed case.

File: server/admin/api.go [hunk 5]
```diff
@@ -156,3 +158,21 @@ func (s *Server) apiAccounts(w http.ResponseWriter, _ *http.Request) {
...
+	acctIDSlice, err := hex.DecodeString(acct)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("could not decode accout id: %v", err), http.StatusBadRequest)
+		return
+	}
+	var acctID account.AccountID
+	copy(acctID[:], acctIDSlice)
```
Comment (chappjc) [inline server/admin/api.go:171]: Should check the length of (acctIDSlice).  Sorry everyone, I don't know why I'm using comments instead of review.

File: server/db/driver/pg/accounts_online_test.go (hunk 2 above)
Comment (JoeGruffins) [reply server/db/driver/pg/accounts_online_test.go:94]: I  added a separate nullifying section.

File: server/db/driver/pg/accounts.go (hunk 1 above)
Comment (JoeGruffins) [reply server/db/driver/pg/accounts.go]: Adding a `Scan` method to `dex.Bytes` looks to has solved the problem. Is this alright? It seems odd that this is needed since scanning when the value is not null used the underlying type's Scan.

File: dex/marshal.go [hunk 6]
```diff
@@ -24,6 +24,18 @@ func (b Bytes) MarshalJSON() ([]byte, error) {
 	return json.Marshal(hex.EncodeToString(b))
 }
 
+// Scan implements the sql.Scanner interface.
+func (b *Bytes) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case []byte:
+		*b = Bytes(src)
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply dex/marshal.go]: Thank you. That saves a bunch of headaches down the road.

**Comment by chappjc** [comment]
For posterity's sake, the results, which could be augmented with an individual account's active swaps at some point:

```
$   dexadm accounts
[
    {
        "accountid": "93e6566e0c4c5c3ac03514fe532f494a7a71b392031c462087bb8c511d532149",
        "pubkey": "0282c421175228136a9f9ed42e7fa63ce4f33900eca27ea1d4675c56e8675b01bf",
        "feeaddress": "SsonWQK6xkLSYm7VttCddHWkWWhETFMdR4Y",
        "feecoin": "f680451275ce09feecb9cc21d829e6e5eeb966dc2c57360f91696ac80742e88a00000000",
        "brokenrule": 0
    }
]
```

```
 $   dexadm account/93e6566e0c4c5c3ac03514fe532f494a7a71b392031c462087bb8c511d532149
{
    "accountid": "93e6566e0c4c5c3ac03514fe532f494a7a71b392031c462087bb8c511d532149",
    "pubkey": "0282c421175228136a9f9ed42e7fa63ce4f33900eca27ea1d4675c56e8675b01bf",
    "feeaddress": "SsonWQK6xkLSYm7VttCddHWkWWhETFMdR4Y",
    "feecoin": "f680451275ce09feecb9cc21d829e6e5eeb966dc2c57360f91696ac80742e88a00000000",
    "brokenrule": 0
}
```
